docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc

```diff
@@ -37,29 +27,106 @@ ifeval::["{build}" == "downstream"]
 MARIADB_IMAGE=registry.redhat.io/rhosp-dev-preview/openstack-mariadb-rhel9:18.0
 endif::[]
-# Replace with your environment's MariaDB Galera cluster VIP and backend IPs:
-SOURCE_MARIADB_IP=172.17.0.2
-declare -A SOURCE_GALERA_MEMBERS
-SOURCE_GALERA_MEMBERS=(
+
+CELLS="default cell1 cell2"
+DEFAULT_CELL_NAME="cell3"
+RENAMED_CELLS="cell1 cell2 $DEFAULT_CELL_NAME"
+
+# The CHARACTER_SET and collation should match the source DB
+# if the do not then it will break foreign key relationships
+# for any tables that are created in the future as part of db sync
+CHARACTER_SET=utf8
+COLLATION=utf8_general_ci
+
+declare -A PODIFIED_DB_ROOT_PASSWORD
+for CELL in $(echo "super $RENAMED_CELLS"); do
+  PODIFIED_DB_ROOT_PASSWORD[$CELL]=$(oc get -o json secret/osp-secret | jq -r .data.DbRootPassword | base64 -d)
+done
+
+declare -A PODIFIED_MARIADB_IP
+for CELL in $(echo "super $RENAMED_CELLS"); do
+  if [ "$CELL" = "super" ]; then
+    PODIFIED_MARIADB_IP[$CELL]=$(oc get svc --selector "mariadb/name=openstack" -ojsonpath='{.items[0].spec.clusterIP}')
+  else
+    PODIFIED_MARIADB_IP[$CELL]=$(oc get svc --selector "mariadb/name=openstack-$CELL" -ojsonpath='{.items[0].spec.clusterIP}')
+  fi
+done
+
+declare -A TRIPLEO_PASSWORDS
+for CELL in $(echo $CELLS); do
+  if [ "$CELL" = "default" ]; then
+    TRIPLEO_PASSWORDS[default]="$HOME/overcloud-passwords.yaml"
+  else
+    # in a split-stack source cloud, it should take a stack-specific passwords file instead
+    TRIPLEO_PASSWORDS[$CELL]="$HOME/overcloud-passwords.yaml"
+  fi
+done
+
+declare -A SOURCE_DB_ROOT_PASSWORD
+for CELL in $(echo $CELLS); do
+  SOURCE_DB_ROOT_PASSWORD[$CELL]=$(cat ${TRIPLEO_PASSWORDS[$CELL]} | grep ' MysqlRootPassword:' | awk -F ': ' '{ print $2; }')
+done
+
+declare -A SOURCE_MARIADB_IP
+SOURCE_MARIADB_IP[default]=*<galera cluster VIP>*
+SOURCE_MARIADB_IP[cell1]=*<galera cell1 cluster VIP>*
+SOURCE_MARIADB_IP[cell2]=*<galera cell2 cluster VIP>*
+# ...
+
+declare -A SOURCE_GALERA_MEMBERS_DEFAULT
+SOURCE_GALERA_MEMBERS_DEFAULT=(
   ["standalone.localdomain"]=172.17.0.100
+  # [...]=...
+)
+declare -A SOURCE_GALERA_MEMBERS_CELL1
+SOURCE_GALERA_MEMBERS_CELL1=(
   # ...
 )
-ifeval::["{build}" != "downstream"]
-SOURCE_DB_ROOT_PASSWORD=$(cat ~/tripleo-standalone-passwords.yaml | grep ' MysqlRootPassword:' | awk -F ': ' '{ print $2; }')
-endif::[]
-ifeval::["{build}" == "downstream"]
-SOURCE_DB_ROOT_PASSWORD=$(cat ~/overcloud-deploy/overcloud/overcloud-passwords.yaml | grep ' MysqlRootPassword:' | awk -F ': ' '{ print $2; }')
-endif::[]
+declare -A SOURCE_GALERA_MEMBERS_CELL2
+SOURCE_GALERA_MEMBERS_CELL2=(
+  # ...
+)
+# ...
 ----
 +
-To get the value to set `SOURCE_MARIADB_IP`, query the puppet-generated configurations in a Controller node:
+Here, `CELLS` and `RENAMED_CELLS` represent changes that are going to be made
+after importing databases: the `default` cell takes a new name from `DEFAULT_CELL_NAME`.
+In a multi-cell adoption scenario, it may retain its original 'default' name as well.
```

Comment:
I'm not sure if it should be in the user-facing docs or if we should consider it too much detail, but is there a simple(-ish) explanation for why we are renaming the cells?

Reply:
This should be users facing. Renaming is needed because for a single cell scenario, the former tripleo cell 'default' needs a rename to a scheme supported by Nova operator.
In a multi-cell adoption scenario, it may retain its original 'default' name, then no renaming of that cell may happen (but we have not tested custom cells names in Nova operator , @gibizer am I right?).
As we have already released the single-cell adoption docs, that would be a proper note to keep it documented